lib/finitomata.ex

```diff
@@ -260,6 +274,9 @@ defmodule Finitomata do
   @type transition_resolution ::
           {:ok, Transition.state(), Finitomata.State.payload()} | {:error, any()}
 
+  @typedoc "The resolution of fork"
+  @type fork_resolution :: {:ok, flow_implementation()}
```

Comment:
Can this fail?

Reply:
We do not expect this to fail. This is a callback called after we reached the _forking state_. If the `on_fork/2` callback fails, it means there is a design error. We handle it internally, but the implementation is not expected to respond with a failure, that’s why this type.